### ISSUE TITLE
AtomicFu version is not resolved leading to build error.

### ISSUE THREAD

**Issue by MarcoSignoretto** [JetBrains/compose-multiplatform#3428]
**Describe the bug**
`compileCommonMainKotlinMetadata` task fails on `Could not find ".../build/kotlinTransformedMetadataLibraries/commonMain/org.jetbrains.kotlinx-atomicfu-0.17.3-nativeInterop-8G5yng.klib"`

I dumped the dependency tree and you can see:
```
org.jetbrains.compose.runtime:runtime:1.5.0-beta01
|    +--- org.jetbrains.kotlinx:atomicfu:0.17.0 -> 0.17.3
|    \--- org.jetbrains.kotlinx:kotlinx-coroutines-core:1.6.4
|         \--- org.jetbrains.kotlinx:atomicfu:0.17.3
```

I was able to make it work forcing atomic fu to 0.21.0 with:

```
subprojects {
    configurations.all {
        resolutionStrategy {
            force("org.jetbrains.kotlinx:atomicfu:0.21.0") 
        }
    }
```

**Affected platforms**
Select one of the platforms below:
- iOS
- ( not sure about other platforms but I guess those might be affected as well )

**Versions**
- Kotlin version*: 1.9.0
- Compose Multiplatform version*: 1.5.0-beta01
- OS version(s)* (required for Desktop and iOS issues): 
- OS architecture (x86 or arm64): both
- JDK (for desktop issues): 

**To Reproduce**
1. clone https://github.com/JetBrains/compose-multiplatform-ios-android-template  at commit `c20faa099552dac7bcc242c72fe49e1f260177e9`
2. ./gradlew build

**Expected behavior**
It should compile and work

**Screenshots**
Before workaround: <img width="1634" alt="Screenshot 2023-07-29 at 10 59 23 AM" src="https://github.com/JetBrains/compose-multiplatform/assets/7922220/a19e8c1b-f80f-4914-bdcd-cd695d72944a">

**Additional context**
* Same issue with Kotlin 1.8.22
* I was able to reproduce this issue also on my project

**Comment by Anirban2022** [comment]
> **Describe the bug** `compileCommonMainKotlinMetadata` task fails on `Could not find ".../build/kotlinTransformedMetadataLibraries/commonMain/org.jetbrains.kotlinx-atomicfu-0.17.3-nativeInterop-8G5yng.klib"`
> 
> I dumped the dependency tree and you can see:
> 
> ```
> org.jetbrains.compose.runtime:runtime:1.5.0-beta01
> |    +--- org.jetbrains.kotlinx:atomicfu:0.17.0 -> 0.17.3
> |    \--- org.jetbrains.kotlinx:kotlinx-coroutines-core:1.6.4
> |         \--- org.jetbrains.kotlinx:atomicfu:0.17.3
> ```
> 
> I was able to make it work forcing atomic fu to 0.21.0 with:
> 
> ```
> subprojects {
>     configurations.all {
>         resolutionStrategy {
>             force("org.jetbrains.kotlinx:atomicfu:0.21.0") 
>         }
>     }
> ```
> 
> **Affected platforms** Select one of the platforms below:
> 
>     * iOS
> 
>     * ( not sure about other platforms but I guess those might be affected as well )
> 
> 
> **Versions**
> 
>     * Kotlin version*: 1.9.0
> 
>     * Compose Multiplatform version*: 1.5.0-beta01
> 
>     * OS version(s)* (required for Desktop and iOS issues):
> 
>     * OS architecture (x86 or arm64): both
> 
>     * JDK (for desktop issues):
> 
> 
> **To Reproduce**
> 
>     1. clone https://github.com/JetBrains/compose-multiplatform-ios-android-template  at commit `c20faa099552dac7bcc242c72fe49e1f260177e9`
> 
>     2. ./gradlew build
> 
> 
> **Expected behavior** It should compile and work
> 
> **Screenshots** Before workaround: <img alt="Screenshot 2023-07-29 at 10 59 23 AM" width="1634" src="https://user-images.githubusercontent.com/7922220/256974250-a19e8c1b-f80f-4914-bdcd-cd695d72944a.png">
> 
> **Additional context**
> 
>     * Same issue with Kotlin 1.8.22
> 
>     * I was able to reproduce this issue also on my project

May I know where did you added it? inside `val commonMain by getting {`??

**Comment by dima-avdeev-jb** [comment]
Faster reproducer, run: `./gradlew :shared:compileCommonMainKotlinMetadata`

**Comment by dima-avdeev-jb** [comment]
Interesting behaviour, if I added to shared module new jvm target:
```Kotlin
kotlin {
   ...
   jvm("workaround")
   ...
}
```
It work's fine

**Comment by Anirban2022** [comment]
I mean to fix it where did you added it.
`subprojects {
    configurations.all {
        resolutionStrategy {
            force("org.jetbrains.kotlinx:atomicfu:0.21.0") 
        }
    }`

**Comment by KevinnZou** [comment]
Are there any updates? Encountered the same issue.

**Comment by dima-avdeev-jb** [comment]
@KevinnZou 
Yeah - it is fixed with Compose Multiplatform version `1.5.10-rc01` and Kotlin `1.9.10`

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-3428">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
